simulator/infra/ports/t/unit_test.cpp

```diff
@@ -22,7 +24,61 @@ namespace ports {
     
         static const int DATA_LIMIT = 5;
         static const uint64 CLOCK_LIMIT = 10;
+
+        class Logic
+        {
+            public:
+                enum
+                {
+                    DATA_FROM_A,
+                    DATA_FROM_B,
+                    DATA_INIT
+                };
+
+                enum 
+                {
+                    IS_FROM_A_READY,
+                    IS_FROM_B_READY,
+                    IS_INIT_READY,
+                    IS_STOP_READY
+                };
+
+                bool check_readiness( uint64 cycle, int check_code, bool is_ready);
```

Comment:
That should be 'const'

Reply:
It changes readiness_logic during an execution, but check_data(..) does not. I have though it would be odd to have only check_data const, but I can make it const, because I don't expect anyone to use the Logic class outside this test. I think if there was an acute need of a Logic class, we should probably think of providing an abstract Logic class which would depend on an arbitrary parameter P. However, there is no test like this.

Reply:
Even if class is not expected to use outside or method is private, ‘const’ prevents from wrong modifications inside the method, like replacing == by =.

John Carmack even proposed a programming language where everything was ‘const’ by default and programmer had to add a ‘mutable’ qualifier if it is required. It is much safer, although annoying.